src/java/org/apache/cassandra/schema/PartitionDenylist.java

```diff
@@ -161,36 +159,35 @@ private boolean readAllHasSufficientNodes()
+                       {
+                           @Override
+                           public DenylistEntry load(final TableId tid) throws Exception
                            {
-                               @Override
-                               public DenylistEntry load(final TableId tid) throws Exception
-                               {
-                                   return getDenylistForTable(tid);
-                               }
-
-                               @Override
-                               public ListenableFuture<DenylistEntry> reload(final TableId tid, final DenylistEntry oldValue)
-                               {
-                                   ListenableFutureTask<DenylistEntry> task = ListenableFutureTask.create(new Callable<DenylistEntry>()
-                                   {
-                                       @Override
-                                       public DenylistEntry call()
-                                       {
-                                           final DenylistEntry newEntry = getDenylistForTable(tid);
-                                           if (newEntry != null)
-                                               return newEntry;
-                                           if (oldValue != null)
-                                               return oldValue;
-                                           return new DenylistEntry();
-                                       }
-                                   });
-                                   executor.execute(task);
-                                   return task;
-                               }
-                           });
+                               return getDenylistForTableFromCQL(tid);
+                           }
+
+                           @Override
+                           public CompletableFuture<DenylistEntry> asyncReload(final TableId tid, final DenylistEntry oldValue, Executor executor)
+                           {
+                               return CompletableFuture.supplyAsync(() -> {
+                                   /* We accept DenylistEntry data in the following precedence:
+                                        1) new data pulled from CQL if we got it correctly,
+                                        2) the old value from the cache if there was some error querying,
+                                        3) an empty record otherwise.
+                                    */
+                                   final DenylistEntry newEntry = getDenylistForTableFromCQL(tid);
+                                   if (newEntry != null)
+                                       return newEntry;
+                                   if (oldValue != null)
+                                       return oldValue;
+                                   return new DenylistEntry();
+                               }, executor);
+                           }
```

Comment:
fyi, you should be able to use the synchronous `reload` method, which `CacheLoader.asyncReload` defaults to wrapping with a `supplyAsync`.

I don't think in either Guava or Caffeine the `oldValue` can every be null, else they delegate to `load(key)` for you (as nothing to "reload").  As the default for `reload(key, oldValue)` is to delegate to `load(key)`, you might not even need this override. In that case you could further simplify even to just `build(this:: getDenylistForTableFromCQL)`.